contracts-periphery/src/UmbraBatchSend.sol

```diff
@@ -24,54 +26,86 @@ contract UmbraBatchSend {
     bytes32 ciphertext;
   }
 
+  mapping(address => uint256) internal totalTransferAmountPerToken;
+  struct TransferSummary {
+    uint256 amount;
+    address tokenAddr;
+  }
+
   error ValueMismatch();
+  error TransferAmmountMismatch();
   event BatchSendExecuted(address indexed sender);
 
   constructor(IUmbra _umbra) {
     umbra = _umbra;
   }
 
   function batchSendEth(uint256 _tollCommitment, SendEth[] calldata _params) external payable {
-    if(msg.value != _sumValueFromEthSends(_tollCommitment, _params)) revert ValueMismatch();
+    if (msg.value != _sumValueFromEthSends(_tollCommitment, _params)) revert ValueMismatch();
     _batchSendEth(_tollCommitment, _params);
     emit BatchSendExecuted(msg.sender);
   }
 
-  function batchSendTokens(uint256 _tollCommitment, SendToken[] calldata _params) external payable {
-    if(msg.value != _tollCommitment * _params.length) revert ValueMismatch();
-    _batchSendTokens(_tollCommitment, _params);
+  function batchSendTokens(
+    uint256 _tollCommitment,
+    SendToken[] calldata _params,
+    TransferSummary[] calldata _transferSummary
+  ) public payable {
+    if (msg.value < _tollCommitment * _params.length) revert ValueMismatch();
+
+    for (uint256 i = 0; i < _params.length; i++) {
+      // Used later to validate total amounts are correctly provided
+      totalTransferAmountPerToken[_params[i].tokenAddr] += _params[i].amount;
+    }
+
+    uint256 _summaryLength = _transferSummary.length;
+    for (uint256 i = 0; i < _summaryLength; i = uncheckedIncrement(i)) {
+      IERC20 _token = IERC20(address(_transferSummary[i].tokenAddr));
+
+      if (totalTransferAmountPerToken[_transferSummary[i].tokenAddr] != _transferSummary[i].amount) {
```

Comment:
We never clear this storage mapping so won't subsequent sends revert with a mismatch here?